app/hooks/android_back_handler.tsx

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {EffectCallback, useEffect} from 'react';
+import {BackHandler} from 'react-native';
+
+import EphemeralStore from '@store/ephemeral_store';
+
+const useAndroidHardwareBackHandler = (componentId: string, callback: EffectCallback) => {
```

Comment:
I like this, really do, will you be submitting another PR to replace this in every component that needs it?

Reply:
Thank you. Yes, I'll do it.   I have created this ticket to tackle the refactoring: https://mattermost.atlassian.net/browse/MM-45022